core/trino-spi/src/main/java/io/trino/spi/expression/Call.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.expression;
+
+import io.trino.spi.connector.CatalogSchemaName;
+import io.trino.spi.type.Type;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+import static java.util.Objects.requireNonNull;
+
+public final class Call
+        extends ConnectorExpression
+{
+    private final Optional<CatalogSchemaName> catalogSchemaName;
+    private final String name;
+    private final List<ConnectorExpression> arguments;
+
+    public Call(
+            Type type,
+            Optional<CatalogSchemaName> catalogSchemaName,
+            String name,
+            List<ConnectorExpression> arguments)
+    {
+        super(type);
+        this.catalogSchemaName = requireNonNull(catalogSchemaName, "catalogSchemaName is null");
+        this.name = requireNonNull(name, "name is null");
+        this.arguments = List.copyOf(requireNonNull(arguments, "arguments is null"));
+    }
+
+    /**
+     * @return the catalog and schema of this function call, or {@link Optional#empty()} if this is a built-in function call
+     */
+    public Optional<CatalogSchemaName> getCatalogSchemaName()
+    {
+        return catalogSchemaName;
+    }
+
+    /**
+     * @return the function's name
+     */
+    public String getName()
```

Comment:
in `io.trino.spi.connector.AggregateFunction#getFunctionName` equivalent field is called "function name"
i suggest to use the same naming here.

also, the fact that we have two name fields -- getCatalogSchemaName and getName -- makes consumption (pattern matching) harder. 
it's easy to write code like
```
if (call.getFunctionName().equals("like")) {
  ...
```

forgetting to check that catalog is empty.

i'd suggest rolling this up into a single attribute.
maybe we need a new class, like `FunctionName`, unless we decide to use `QualifiedName` here (despite https://github.com/trinodb/trino/pull/7994/files#r705614641)

Reply:
Created a new class

Reply:
> maybe we need a new class, like FunctionName, unless we decide to use QualifiedName here (despite https://github.com/trinodb/trino/pull/7994/files#r705614641)

Not sure what that link is referencing -- it just takes me to the top of the PR. I agree that a holder for catalog/schema/name is needed, but QualifiedName can't be used because 1) it's not in the SPI and 2) has undesirable behavior around case-normalization of names. 

The name here has to satisfy a couple of properties:
* It should be interpreted as "already-normalized", so it shouldn't do any case conversions
* For the time being, existing functions will have a missing catalog/schema name for backward compatibility. In the future, once https://github.com/trinodb/trino/issues/8 is finalized, we may change this and mount all built in functions under a well-known catalog/schema that's present in CURRENT_PATH. In general, having an optional catalog/schema is confusing, as it can be interpreted as "resolve the name based on the current catalog and schema" in some contexts.

Reply:
> In general, having an optional catalog/schema is confusing, as it can be interpreted as "resolve the name based on the current catalog and schema" in some contexts.

The currently existing functions have no catalog/schema, so we cannot provide one.
Unless we can agree on the catalog/schema name they will eventually get and use these today within connector expression.

Having no catalog/schema field for now is not an option, since adding it later would be a breaking change for consumers.

The `FunctionName` class added in this PR does the job.